### PR TITLE
:sparkles: promote IPAM types to v1beta1

### LINKED ISSUE #8694: Consider moving IPAM provider out of experimental
- What are the requirements for this provider to move to v1beta1 types / guarantees?
- Are there long standing issues?
- What would be the next logical step from the current state?

cc @schrej

### PR DESCRIPTION
**Pull request by rvanderp3**
**What this PR does / why we need it**: 
Promotes IPAM types to v1beta1.  

**Which issue(s) this PR fixes**:
Fixes #8694

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
Thanks for taking this on!

/ok-to-test

**Comment by rvanderp3** [comment]
/hold

placing hold while test failures are reviewed

**Comment by rvanderp3** [comment]
> @rvanderp3 We should add the conversion webhooks as part of this PR, feel free to reach out if you have questions

Sure thing, thanks!

**Comment by rvanderp3** [comment]
/assign @vincepri 

@vincepri this is ready for review. would you mind taking a look when you get a chance?

**Comment by killianmuldoon** [comment]
/hold

Just want to make sure this gets review from the right folks before its merged.

File: exp/ipam/api/v1alpha1/conversion.go
Comment (schrej) [inline exp/ipam/api/v1alpha1/conversion.go:46]: Do we really need to call MarshalData here if we aren't using it?

**Comment by rvanderp3** [comment]
/test pull-cluster-api-test-main

there seems to be a flake, i'll investigate and open an issue on this

File: exp/ipam/api/v1alpha1/conversion.go
Comment (rvanderp3) [reply exp/ipam/api/v1alpha1/conversion.go]: no, we do not.  i just removed it.

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**
